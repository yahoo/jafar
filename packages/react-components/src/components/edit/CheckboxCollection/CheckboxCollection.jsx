/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import React from 'react';
import PropTypes from 'prop-types';
import InternalCheckboxCollection from './internal/CheckboxCollection';

/**
 * Represent an array of any type
 * 
 * Import <a target="_blank" 
 href="https://github.com/yahoo/jafar/blob/master/packages/react-components/src/components/edit/CheckboxCollection/CheckboxCollection.jsx">
 CheckboxCollection</a> from '@jafar-org/react-components/edit/CheckboxCollection'
 */
export default class CheckboxCollection extends React.Component {
  static propTypes = {
    value: PropTypes.array,
    state: PropTypes.shape({
      items: PropTypes.array,
      search: PropTypes.object,
      inline: PropTypes.bool,
    }),
    disabled: PropTypes.bool,
    onValueChange: PropTypes.func.isRequired,
    onStateChange: PropTypes.func,
  };

  static defaultProps = {
    value: [],
    state: {
      items: [],
      search: {
        value: '',
      },
      inline: false,
    },
    disabled: false,
  };

  render() {
    return (
      <InternalCheckboxCollection
        search={this.props.state.search}
        items={this.props.state.items}
        inline={this.props.state.inline}
        value={this.props.value}
        disabled={this.props.disabled}
        onChange={this.props.onValueChange}
        onSearchChange={this.onSearchChange}
      />
    );
  }

  onSearchChange = (value) => {
    this.props.onStateChange({ ...this.props.state, search: { ...this.props.state.search, value } });
  }
}
