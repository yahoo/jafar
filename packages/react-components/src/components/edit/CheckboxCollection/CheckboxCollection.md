<h3>Examples</h3>

<h4>Usage in jafar form</h4>

```javascript
const Form = require('@jafar/react-form/Form').default;
const Field = require('@jafar/react-form/Field').default;

const model = {
  id: 'simple',
  fields: {
    friends: {
      label: 'Friends',
      path: 'friends',
      component: {
        name: 'CheckboxCollection',
        state: {
            items: [{
                value: '1',
                label: 'Ross Geller'
            }, {
                value: '2',
                label: 'Monica Geller'
            }, {
                value: '3',
                label: 'Rachel Green'
            }, {
                value: '4',
                label: 'Chandler Bing'
            }, {
                value: '5',
                label: 'Joey Tribbiani'
            }, {
                value: '6',
                label: 'Phoebe Buffay'
            }]
        }
      },
      required: true,
      disableTerm: {
          name: 'disableMe',
      },
      excludeTerm: {
          name: 'excludeMe',
      },
    },
  },
  data: {
      friends: ['2', '3'], // remove to see initial value undefined, and required error
  },
};

const resources = {
    components: {
        CheckboxCollection: {
            renderer: CheckboxCollection,
        },
    },
    terms: {
      disableMe: { func: () => false }, // convert to true to see the field disabled
      excludeMe: { func: () => false }, // convert to true to see the field excluded
  },
};

<Form model={model} resources={resources}>
    <Field id='friends' />
</Form>
```

<h4>Simple</h4>

```javascript
initialState = { 
    value: ['2', '3'],
    state: {
        items: [{
            value: '1',
            label: 'Ross Geller'
        }, {
            value: '2',
            label: 'Monica Geller'
        }, {
            value: '3',
            label: 'Rachel Green'
        }, {
            value: '4',
            label: 'Chandler Bing'
        }, {
            value: '5',
            label: 'Joey Tribbiani'
        }, {
            value: '6',
            label: 'Phoebe Buffay'
        }]
    }
};

<CheckboxCollection
value={state.value}
state={state.state}
onValueChange={(value) => { 
    setState({ value });
    console.log(`onValueChange: ${value}`);
}}
/>
```


<h4>Disabled</h4>

```javascript
initialState = { 
    value: [],
    state: {
        items: [{
            value: '1',
            label: 'Ross Geller'
        }, {
            value: '2',
            label: 'Monica Geller'
        }, {
            value: '3',
            label: 'Rachel Green'
        }, {
            value: '4',
            label: 'Chandler Bing'
        }, {
            value: '5',
            label: 'Joey Tribbiani'
        }, {
            value: '6',
            label: 'Phoebe Buffay'
        }]
    }
};

<CheckboxCollection
value={state.value}
state={state.state}
disabled={true}
onValueChange={(value) => { 
    setState({ value });
    console.log(`onValueChange: ${value}`);
}}
/>

```

<h4>Inline</h4>

```javascript
initialState = { 
    value: [],
    state: {
        inline: true,
        items: [{
            value: '1',
            label: 'Ross Geller'
        }, {
            value: '2',
            label: 'Monica Geller'
        }, {
            value: '3',
            label: 'Rachel Green'
        }, {
            value: '4',
            label: 'Chandler Bing'
        }, {
            value: '5',
            label: 'Joey Tribbiani'
        }, {
            value: '6',
            label: 'Phoebe Buffay'
        }]
    }
};

<CheckboxCollection
value={state.value}
state={state.state}
onValueChange={(value) => { 
    setState({ value });
    console.log(`onValueChange: ${value}`);
}}
/>
```

<h4>Search</h4>

```javascript
const allItems = [{
    value: '1',
    label: 'Ross Geller'
}, {
    value: '2',
    label: 'Monica Geller'
}, {
    value: '3',
    label: 'Rachel Green'
}, {
    value: '4',
    label: 'Chandler Bing'
}, {
    value: '5',
    label: 'Joey Tribbiani'
}, {
    value: '6',
    label: 'Phoebe Buffay'
}];

initialState = { 
    value: [],
    state: {
        search: {
            value: '',
            placeholder: 'Search'
        },
        items: allItems,
    }
};

<CheckboxCollection
value={state.value}
state={state.state}
onValueChange={(value) => { 
    setState({ value });
    console.log(`onValueChange: ${value}`);
}}
onStateChange={(state) => { 
    state.items = allItems.filter(x => { 
        return x.label.toLowerCase().indexOf(state.search.value.toLowerCase()) > -1;
    });

    setState({ state });
    console.log(`onStateChange: ${state}`);
}}
/>
```

<h4>Async Search</h4>

```javascript
const allItems = [{
    value: '1',
    label: 'Ross Geller'
}, {
    value: '2',
    label: 'Monica Geller'
}, {
    value: '3',
    label: 'Rachel Green'
}, {
    value: '4',
    label: 'Chandler Bing'
}, {
    value: '5',
    label: 'Joey Tribbiani'
}, {
    value: '6',
    label: 'Phoebe Buffay'
}];

initialState = { 
    value: [],
     state: {
        search: {
            value: '',
            placeholder: 'Search'
        },
        items: allItems,
    }
};

<CheckboxCollection
value={state.value}
state={state.state}
onValueChange={(value) => { 
    setState({ value });
    console.log(`onValueChange: ${value}`);
}}
onStateChange={(state) => { 
    setState({ state }, function() {
        setTimeout(() => {
        const items = allItems.filter(x => { 
        return x.label.toLowerCase().indexOf(this.state.state.search.value.toLowerCase()) > -1;
        });

        const updatedState = Object.assign({}, this.state.state);
        updatedState.items = items;
        setState({ state: updatedState });
        }, 200);
    });
}}
/>
```