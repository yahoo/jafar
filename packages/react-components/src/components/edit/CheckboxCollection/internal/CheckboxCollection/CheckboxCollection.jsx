/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */
 
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { isEqual, cloneDeep, noop } from 'lodash';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';
import SearchInput from '../SearchInput';

const itemShape = {
  label: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
};

const searchShape = {
  value: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
};

export default class CheckboxCollection extends React.Component {
  static propTypes = {
    value: PropTypes.arrayOf(PropTypes.string),
    items: PropTypes.arrayOf(PropTypes.shape(itemShape)),
    disabled: PropTypes.bool,
    inline: PropTypes.bool,
    search: PropTypes.shape(searchShape),
    onChange: PropTypes.func,
    onSearchChange: PropTypes.func,
  };

  static defaultProps = {
    value: [],
    items: [],
    disabled: false,
    inline: false,
    search: undefined,
    onChange: noop,
    onSearchChange: noop,
  };

  shouldComponentUpdate(nextProps) {
    return !isEqual(this.props, nextProps);
  }

  onChange = (e, checked) => {
    const itemValue = e.target.value;
    const value = cloneDeep(this.props.value);
    if (checked) {
      value.push(itemValue);
    } else {
      value.splice(value.indexOf(itemValue), 1);
    }
    this.props.onChange(value);
  };

  onSearchChange = value => {
    this.props.onSearchChange(value);
  };

  render() {
    const Item = this.props.inline
      ? styled.div`
          display: inline-block;
        `
      : styled.div``;

    return (
      <React.Fragment>
        {this.props.search && (
          <SearchInput
            value={this.props.search.value}
            placeholder={this.props.search.placeholder}
            disabled={this.props.disabled}
            onChange={this.onSearchChange}
          />
        )}
        {this.props.items.map(item => {
          const checked = this.props.value.indexOf(item.value) > -1;
          return (
            <Item key={item.label}>
              <FormControlLabel
                control={
                  <Checkbox
                    onChange={this.onChange}
                    value={item.value}
                    checked={checked}
                    disabled={this.props.disabled}
                  />
                }
                label={item.label}
              />
            </Item>
          );
        })}
      </React.Fragment>
    );
  }
}
