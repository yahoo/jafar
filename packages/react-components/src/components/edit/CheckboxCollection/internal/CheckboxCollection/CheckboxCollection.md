<h4>Simple</h4>

```javascript
const CheckboxCollection = require('CheckboxCollection').default;
initialState = { 
    value: [],
};
const items = [{
    value: '1',
    label: 'Ross Geller'
}, {
    value: '2',
    label: 'Monica Geller'
}, {
    value: '3',
    label: 'Rachel Green'
}, {
    value: '4',
    label: 'Chandler Bing'
}, {
    value: '5',
    label: 'Joey Tribbiani'
}, {
    value: '6',
    label: 'Phoebe Buffay'
}];

<CheckboxCollection
value={state.value}
items={items}
onChange={(value) => { 
    setState({ value });
    console.log(`onChange: ${value}`);
}}
/>
```

<h4>Disabled</h4>

```javascript
const CheckboxCollection = require('CheckboxCollection').default;
initialState = { 
    value: [],
};
const items = [{
    value: '1',
    label: 'Ross Geller'
}, {
    value: '2',
    label: 'Monica Geller'
}, {
    value: '3',
    label: 'Rachel Green'
}, {
    value: '4',
    label: 'Chandler Bing'
}, {
    value: '5',
    label: 'Joey Tribbiani'
}, {
    value: '6',
    label: 'Phoebe Buffay'
}];

<CheckboxCollection
value={state.value}
items={items}
disabled={true}
onChange={(value) => { 
    setState({ value });
    console.log(`onChange: ${value}`);
}}
/>

```

<h4>Inline</h4>

```javascript
const CheckboxCollection = require('CheckboxCollection').default;
initialState = { 
    value: [],
};

const items = [{
    value: '1',
    label: 'Ross Geller'
}, {
    value: '2',
    label: 'Monica Geller'
}, {
    value: '3',
    label: 'Rachel Green'
}, {
    value: '4',
    label: 'Chandler Bing'
}, {
    value: '5',
    label: 'Joey Tribbiani'
}, {
    value: '6',
    label: 'Phoebe Buffay'
}];

<CheckboxCollection
value={state.value}
items={items}
inline={true}
onChange={(value) => { 
    setState({ value });
    console.log(`onChange: ${value}`);
}}
/>
```

<h4>Search</h4>

```javascript
const CheckboxCollection = require('CheckboxCollection').default;
const allItems = [{
    value: '1',
    label: 'Ross Geller'
}, {
    value: '2',
    label: 'Monica Geller'
}, {
    value: '3',
    label: 'Rachel Green'
}, {
    value: '4',
    label: 'Chandler Bing'
}, {
    value: '5',
    label: 'Joey Tribbiani'
}, {
    value: '6',
    label: 'Phoebe Buffay'
}];

initialState = { 
    value: [],
    search: {
        value: '',
        placeholder: 'Search'
    },
    items: allItems,
};

<CheckboxCollection
value={state.value}
items={state.items}
search={state.search}
onChange={(value) => { 
    setState({ value });
    console.log(`onChange: ${value}`);
}}
onSearchChange={(value) => { 
    const search = Object.assign({}, state.search);
    Object.assign(search, {value});
    const items = allItems.filter(x => { 
            return x.label.toLowerCase().indexOf(value.toLowerCase()) > -1;
        });
    setState({ search, items });

    console.log(`onSearchChange: ${value}`);
}}
/>
```

<h4>Async Search</h4>

```javascript
const CheckboxCollection = require('CheckboxCollection').default;
const allItems = [{
    value: '1',
    label: 'Ross Geller'
}, {
    value: '2',
    label: 'Monica Geller'
}, {
    value: '3',
    label: 'Rachel Green'
}, {
    value: '4',
    label: 'Chandler Bing'
}, {
    value: '5',
    label: 'Joey Tribbiani'
}, {
    value: '6',
    label: 'Phoebe Buffay'
}];

initialState = { 
    value: [],
    search: {
        value: '',
        placeholder: 'Search'
    },
    items: allItems,
};

<CheckboxCollection
value={state.value}
items={state.items}
search={state.search}
onChange={(value) => { 
    setState({ value });
    console.log(`onChange: ${value}`);
}}
onSearchChange={(value) => { 
    const search = Object.assign({}, state.search);
    Object.assign(search, {value});
    setState({ 
        search,
        items: [],
    });

    setTimeout(() => {
        const items = allItems.filter(x => { 
            return x.label.toLowerCase().indexOf(value.toLowerCase()) > -1;
        });

        setState({ items });
    }, 300); 

    console.log(`onSearchChange: ${value}`);
}}
/>
```