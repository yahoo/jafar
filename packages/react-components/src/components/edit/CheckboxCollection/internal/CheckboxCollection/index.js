import CheckboxCollection from './CheckboxCollection';
export default CheckboxCollection;
