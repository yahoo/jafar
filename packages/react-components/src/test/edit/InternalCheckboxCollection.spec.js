/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import React from 'react';
import { shallow, mount } from 'enzyme';
import CheckboxCollection from '../../edit/CheckboxCollection/internal/CheckboxCollection/index';

describe('<CheckboxCollection />', () => {
  let component;
  let onChangeSpy;
  let onSearchChangeSpy;
  const value = ['1', '2'];
  const items = [
    {
      value: '1',
      label: 'Ross Geller',
    },
    {
      value: '2',
      label: 'Monica Geller',
    },
    {
      value: '3',
      label: 'Rachel Green',
    },
    {
      value: '4',
      label: 'Chandler Bing',
    },
    {
      value: '5',
      label: 'Joey Tribbiani',
    },
    {
      value: '6',
      label: 'Phoebe Buffay',
    },
  ];
  let disabled;
  let inline;
  let search;

  beforeEach(() => {
    disabled = false;
    inline = false;
    search = undefined;
    onChangeSpy = jest.fn();
    onSearchChangeSpy = jest.fn();
  });

  describe('should render provided data', () => {
    it('checkboxes only', () => {
      component = shallow(getComponent(value, items, disabled, inline, search, onChangeSpy, onSearchChangeSpy));
      expect(component).toMatchSnapshot();
    });

    it('checkboxes inline', () => {
      inline = true;
      component = shallow(getComponent(value, items, disabled, inline, search, onChangeSpy, onSearchChangeSpy));
      expect(component).toMatchSnapshot();
    });

    it('checkboxes disabled', () => {
      disabled = true;
      component = shallow(getComponent(value, items, disabled, inline, search, onChangeSpy, onSearchChangeSpy));
      expect(component).toMatchSnapshot();
    });

    it('checkboxes with search', () => {
      search = { value: 'geller', placeholder: 'Search' };
      component = shallow(getComponent(value, items, disabled, inline, search, onChangeSpy, onSearchChangeSpy));
      expect(component).toMatchSnapshot();
    });
  });

  describe('shouldComponentUpdate check', () => {
    it('return true - when nextProps are not equals to currProps', () => {
      component = shallow(getComponent(value, items, disabled, inline, search, onChangeSpy, onSearchChangeSpy));
      const nextProps = {};
      expect(component.instance().shouldComponentUpdate(nextProps)).toBeTruthy();
    });

    it('return false - when nextProps are equals to currProps', () => {
      component = shallow(getComponent(value, items, disabled, inline, search, onChangeSpy, onSearchChangeSpy));
      const nextProps = {
        value, items, disabled, inline, search, onChange: onChangeSpy, onSearchChange: onSearchChangeSpy,
      };
      expect(component.instance().shouldComponentUpdate(nextProps)).toBeFalsy();
    });
  });

  describe('should trigger', () => {
    beforeEach(() => {
      search = { value: '', placeholder: 'Search' };
      component = mount(getComponent(value, items, disabled, inline, search, onChangeSpy, onSearchChangeSpy));
    });

    it('onSearchChange callback on search value change', () => {
      const newValue = 'banana';
      component
        .find('input[type="search"]')
        .simulate('change', { target: { value: newValue } });
      expect(onSearchChangeSpy).toHaveBeenCalledWith(newValue);
    });

    it('onChange callback on checkbox checked', () => {
      component.find('[value="3"] input[type="checkbox"]').simulate('change', {
        target: { value: '3', checked: true },
      });
      expect(onChangeSpy).toHaveBeenCalledWith(['1', '2', '3']);
    });

    it('onChange callback on checkbox unchecked', () => {
      component.find('[value="3"] input[type="checkbox"]').simulate('change', {
        target: { value: '2', checked: false },
      });
      expect(onChangeSpy).toHaveBeenCalledWith(['1']);
    });
  });

  function getComponent(value, items, disabled, inline, search, onChange, onSearchChange) {
    return (
      <CheckboxCollection value={value} items={items} disabled={disabled} inline={inline} search={search}
        onChange={onChange} onSearchChange={onSearchChange} />);
  }
});
