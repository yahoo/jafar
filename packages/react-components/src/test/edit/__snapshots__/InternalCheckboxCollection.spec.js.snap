// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckboxCollection /> should render provided data checkboxes disabled 1`] = `
<Fragment>
  <styled.div
    key="Ross Geller"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={
        <ForwardRef(WithStyles)
          checked={true}
          disabled={true}
          onChange={[Function]}
          value="1"
        />
      }
      label="Ross Geller"
    />
  </styled.div>
  <styled.div
    key="Monica Geller"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={
        <ForwardRef(WithStyles)
          checked={true}
          disabled={true}
          onChange={[Function]}
          value="2"
        />
      }
      label="Monica Geller"
    />
  </styled.div>
  <styled.div
    key="Rachel Green"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={
        <ForwardRef(WithStyles)
          checked={false}
          disabled={true}
          onChange={[Function]}
          value="3"
        />
      }
      label="Rachel Green"
    />
  </styled.div>
  <styled.div
    key="Chandler Bing"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={
        <ForwardRef(WithStyles)
          checked={false}
          disabled={true}
          onChange={[Function]}
          value="4"
        />
      }
      label="Chandler Bing"
    />
  </styled.div>
  <styled.div
    key="Joey Tribbiani"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={
        <ForwardRef(WithStyles)
          checked={false}
          disabled={true}
          onChange={[Function]}
          value="5"
        />
      }
      label="Joey Tribbiani"
    />
  </styled.div>
  <styled.div
    key="Phoebe Buffay"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={
        <ForwardRef(WithStyles)
          checked={false}
          disabled={true}
          onChange={[Function]}
          value="6"
        />
      }
      label="Phoebe Buffay"
    />
  </styled.div>
</Fragment>
`;

exports[`<CheckboxCollection /> should render provided data checkboxes inline 1`] = `
<Fragment>
  <styled.div
    key="Ross Geller"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={
        <ForwardRef(WithStyles)
          checked={true}
          disabled={false}
          onChange={[Function]}
          value="1"
        />
      }
      label="Ross Geller"
    />
  </styled.div>
  <styled.div
    key="Monica Geller"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={
        <ForwardRef(WithStyles)
          checked={true}
          disabled={false}
          onChange={[Function]}
          value="2"
        />
      }
      label="Monica Geller"
    />
  </styled.div>
  <styled.div
    key="Rachel Green"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={
        <ForwardRef(WithStyles)
          checked={false}
          disabled={false}
          onChange={[Function]}
          value="3"
        />
      }
      label="Rachel Green"
    />
  </styled.div>
  <styled.div
    key="Chandler Bing"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={
        <ForwardRef(WithStyles)
          checked={false}
          disabled={false}
          onChange={[Function]}
          value="4"
        />
      }
      label="Chandler Bing"
    />
  </styled.div>
  <styled.div
    key="Joey Tribbiani"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={
        <ForwardRef(WithStyles)
          checked={false}
          disabled={false}
          onChange={[Function]}
          value="5"
        />
      }
      label="Joey Tribbiani"
    />
  </styled.div>
  <styled.div
    key="Phoebe Buffay"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={
        <ForwardRef(WithStyles)
          checked={false}
          disabled={false}
          onChange={[Function]}
          value="6"
        />
      }
      label="Phoebe Buffay"
    />
  </styled.div>
</Fragment>
`;

exports[`<CheckboxCollection /> should render provided data checkboxes only 1`] = `
<Fragment>
  <styled.div
    key="Ross Geller"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={
        <ForwardRef(WithStyles)
          checked={true}
          disabled={false}
          onChange={[Function]}
          value="1"
        />
      }
      label="Ross Geller"
    />
  </styled.div>
  <styled.div
    key="Monica Geller"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={
        <ForwardRef(WithStyles)
          checked={true}
          disabled={false}
          onChange={[Function]}
          value="2"
        />
      }
      label="Monica Geller"
    />
  </styled.div>
  <styled.div
    key="Rachel Green"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={
        <ForwardRef(WithStyles)
          checked={false}
          disabled={false}
          onChange={[Function]}
          value="3"
        />
      }
      label="Rachel Green"
    />
  </styled.div>
  <styled.div
    key="Chandler Bing"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={
        <ForwardRef(WithStyles)
          checked={false}
          disabled={false}
          onChange={[Function]}
          value="4"
        />
      }
      label="Chandler Bing"
    />
  </styled.div>
  <styled.div
    key="Joey Tribbiani"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={
        <ForwardRef(WithStyles)
          checked={false}
          disabled={false}
          onChange={[Function]}
          value="5"
        />
      }
      label="Joey Tribbiani"
    />
  </styled.div>
  <styled.div
    key="Phoebe Buffay"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={
        <ForwardRef(WithStyles)
          checked={false}
          disabled={false}
          onChange={[Function]}
          value="6"
        />
      }
      label="Phoebe Buffay"
    />
  </styled.div>
</Fragment>
`;

exports[`<CheckboxCollection /> should render provided data checkboxes with search 1`] = `
<Fragment>
  <WithStyles(SearchInput)
    disabled={false}
    onChange={[Function]}
    placeholder="Search"
    value="geller"
  />
  <styled.div
    key="Ross Geller"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={
        <ForwardRef(WithStyles)
          checked={true}
          disabled={false}
          onChange={[Function]}
          value="1"
        />
      }
      label="Ross Geller"
    />
  </styled.div>
  <styled.div
    key="Monica Geller"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={
        <ForwardRef(WithStyles)
          checked={true}
          disabled={false}
          onChange={[Function]}
          value="2"
        />
      }
      label="Monica Geller"
    />
  </styled.div>
  <styled.div
    key="Rachel Green"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={
        <ForwardRef(WithStyles)
          checked={false}
          disabled={false}
          onChange={[Function]}
          value="3"
        />
      }
      label="Rachel Green"
    />
  </styled.div>
  <styled.div
    key="Chandler Bing"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={
        <ForwardRef(WithStyles)
          checked={false}
          disabled={false}
          onChange={[Function]}
          value="4"
        />
      }
      label="Chandler Bing"
    />
  </styled.div>
  <styled.div
    key="Joey Tribbiani"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={
        <ForwardRef(WithStyles)
          checked={false}
          disabled={false}
          onChange={[Function]}
          value="5"
        />
      }
      label="Joey Tribbiani"
    />
  </styled.div>
  <styled.div
    key="Phoebe Buffay"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={
        <ForwardRef(WithStyles)
          checked={false}
          disabled={false}
          onChange={[Function]}
          value="6"
        />
      }
      label="Phoebe Buffay"
    />
  </styled.div>
</Fragment>
`;
